src/devices/slacky_diy.ts

```diff
@@ -0,0 +1,791 @@
+            {
+                cluster: "seMetering",
+                type: ["attributeReport", "readResponse"],
+                convert: (model, msg, publish, options, meta) => {
+                    const result: KeyValueAny = {};
+                    if (msg.data.currentTier4SummDelivered !== undefined) {
+                        const data = msg.data.currentTier4SummDelivered;
+                        result.tariff4 = (Number.parseInt(data) / energyDivisor) * energyMultiplier;
+                    }
+                    return result;
+                },
+            },
+            {
+                cluster: "seMetering",
+                type: ["attributeReport", "readResponse"],
+                convert: (model, msg, publish, options, meta) => {
+                    const result: KeyValueAny = {};
+                    if (msg.data.currentSummDelivered !== undefined) {
+                        const data = msg.data.currentSummDelivered;
+                        result.tariff_summ = (Number.parseInt(data) / energyDivisor) * energyMultiplier;
+                    }
+                    return result;
+                },
+            },
+        ];
+        return {
+            exposes,
+            fromZigbee,
+            toZigbee,
+            isModernExtend: true,
+        };
+    },
+    powerSupplyParam: (): ModernExtend => {
+        const exposes: Expose[] = [
+            e.numeric("voltage", ea.STATE_GET).withUnit("V").withDescription("Voltage"),
+            e.numeric("current", ea.STATE_GET).withUnit("A").withDescription("Current"),
+            e.numeric("power", ea.STATE_GET).withUnit("kW").withDescription("Power"),
+        ];
+        const toZigbee: Tz.Converter[] = [
+            {
+                key: ["voltage"],
+                convertGet: async (entity, key, meta) => {
+                    await entity.read("haElectricalMeasurement", ["rmsVoltage", "acVoltageMultiplier", "acVoltageDivisor"]);
+                },
+                convertSet: async (entity, key, value, meta) => {
+                    return await null;
+                },
+            },
+            {
+                key: ["current"],
+                convertGet: async (entity, key, meta) => {
+                    await entity.read("haElectricalMeasurement", ["instantaneousLineCurrent", "acCurrentMultiplier", "acCurrentDivisor"]);
+                },
+                convertSet: async (entity, key, value, meta) => {
+                    return await null;
+                },
+            },
+            {
+                key: ["power"],
+                convertGet: async (entity, key, meta) => {
+                    await entity.read("haElectricalMeasurement", ["apparentPower", "acPowerMultiplier", "acPowerDivisor"]);
+                },
+                convertSet: async (entity, key, value, meta) => {
+                    return await null;
+                },
+            },
+        ];
+        const fromZigbee: Fz.Converter[] = [
```

Comment:
Can't you use the `m.electricityMeter()` instead? That will prevent a lot of code duplication.